### PR TITLE
Improvement of SQL generator

### PR DESCRIPTION
**Pull request by you06**
Signed-off-by: you06 <you1474600@gmail.com>

- generator interface
- support online DDL mode
- configuration check duration
- configuration init table count
- default value for int/float/varchar/timestamp/datetime
- add a data compare tool which can find difference by uuid
- support general log in config

### CODE REVIEW

File: go-sqlsmith/dml_sql.go [hunk 1]
```diff
@@ -12,25 +12,26 @@ import (
...
+// InsertStmt implement insert statement from AST
+func (s *SQLSmith) InsertStmt(fn bool) (string, string, error) {
 	tree := s.insertStmt()
 	return s.Walk(tree)
 }
 
-// InsertStmt make random insert statement SQL
-func (s *SQLSmith) InsertStmt(fn bool) (string, error) {
+// InsertStmtStr make random insert statement SQL
+func (s *SQLSmith) InsertStmtStr(fn bool) (string, string, error) {
```
Comment (mahjonp) [inline go-sqlsmith/dml_sql.go]: Seems there is no function will call InsertStmtStr?

File: go-sqlsmith/stateflow/schema.go [hunk 2]
```diff
@@ -59,22 +59,34 @@ func (s *StateFlow) randTableFromTable(table *types.Table, newName bool, fn bool
...
-		if index == k {
-			for len(newTable.Columns) == 0 {
-				newTable = s.randTableFromTable(table, newName, fn)
-			}
-			return newTable
+func (s *StateFlow) randTable(newName bool, fn bool, online bool) (*types.Table) {
+	tables := []*types.Table{}
+	for _, table := range s.db.Tables {
+		// get online tables
+		if !online || !s.db.Online || table.Online {
```
Comment (mahjonp) [inline go-sqlsmith/stateflow/schema.go]: I'm confused, db.Online and table.Online wasn't always set to same boolean value? 😶

File: go-sqlsmith/types/table.go [hunk 3]
```diff
@@ -7,12 +7,16 @@ import (
...
-	Type string
-	Columns map[string]*Column
-	Indexes []string
+	Type        string
+	Columns     map[string]*Column
+	Indexes     []string
+	// Online is for self obtain
+	Online      bool
+	// OnlineOther is for other instances obtain
+	OnlineOther bool
```
Comment (mahjonp) [inline go-sqlsmith/types/table.go]: what are the meaning of `Online` and `OnlineOther`? 😆

**Comment by you06** [comment]
@mahjonp I'll add some comments later, and this PR is not ready in this time.

File: go-sqlsmith/dml_sql.go (hunk 1 above)
Comment (you06) [reply go-sqlsmith/dml_sql.go]: Yes, but there are 2 ways of generate insert statement, from AST and building from string directly. Just keep both of them here.

**Comment by zhouqiang-cl** [comment]
Can we review this pr now @you06

File: go-sqlsmith/data_gen_test.go [hunk 4]
```diff
@@ -16,13 +16,13 @@ package sqlsmith
 import "testing"
 
 func TestSQLSmith_GenData(t *testing.T) {
```
Comment (mapleFU) [inline go-sqlsmith/data_gen_test.go]: Why this is leaved but all commented?

**Comment by you06** [comment]
@zhouqiang-cl Still some code tidy up works to do, there will be some updates tomorrow.

File: go-sqlsmith/data_gen_test.go (hunk 4 above)
Comment (you06) [reply go-sqlsmith/data_gen_test.go]: There should be, please review later.

File: go-sqlsmith/ddl_ast.go [hunk 5]
```diff
@@ -26,9 +26,9 @@ func (s *SQLSmith) createTableStmt() ast.Node {
 		Constraints: []*ast.Constraint{},
 		Options: []*ast.TableOption{},
 	}
-	if util.Rd(4) == 0 {
-		createTableNode.Partition = s.partitionStmt()
-	}
+	// if util.Rd(4) == 0 {
+	// 	createTableNode.Partition = s.partitionStmt()
```
Comment (mahjonp) [inline go-sqlsmith/ddl_ast.go]: go-sqlsmith should add more switches in future.

File: go-sqlsmith/stateflow/schema.go (hunk 2 above)
Comment (you06) [reply go-sqlsmith/stateflow/schema.go]: They are not, this is for avoiding online DDL, when start a txn, db.Online and some tables will be set to true, and the other tables has Online value false, then the go-sqlsmith should generate DDL SQLs only about these tables during this txn.

File: go-sqlsmith/types/table.go (hunk 3 above)
Comment (you06) [reply go-sqlsmith/types/table.go]: `Online` means this table is online and will be manipulated in a txn, `OnlineOther` means this table is manipulated in other transactions which should not be a DDL table.

**Pull request merged**